apps/client/src/app/lobby/[roomId]/page.tsx

```diff
@@ -0,0 +1,179 @@
+'use client';
+
+import Image from 'next/image';
+import { redirect, useParams } from 'next/navigation';
+import { useSession } from 'next-auth/react';
+import React, { useEffect, useRef, useState } from 'react';
+
+import { socket } from '@/socket';
+//need to change User to IUser and retrive the data from the server
+
+type User =
+  | {
+      name?: string | null | undefined;
+      email?: string | null | undefined;
+      image?: string | null | undefined;
+    }
+  | undefined;
+
+export default function LobbyPage() {
+  const { roomId } = useParams();
+  const { data: session } = useSession({
+    required: false,
+  });
+  const user = session?.user ?? ({ name: 'Guest' } as User);
+  const [ready, setReady] = useState(false);
+  //test map
+  const [players, setPlayers] = useState<string[]>([]);
+  const [isGameStarting, setIsGameStarting] = useState(false);
+  const hasJoined = useRef(false);
+
+  useEffect(() => {
+    if (!user || !socket || !roomId) return;
+
+    if (!hasJoined.current) {
+      socket.emit('join_lobby', { username: user.name, roomId });
+      console.log(`user ${user?.name} joined room-${roomId}`);
+      hasJoined.current = true; // Mark as joined
+    }
+    if (players.length >= 2) {
+      setReady(true);
+    }
+
+    socket.on('update_players', (playerList: string[]) => {
+      setPlayers(playerList);
+    });
+    socket.on('start_game', () => {
+      setIsGameStarting(true);
+      redirect(`/game/${roomId}`);
+    });
+
+    return () => {
+      socket.emit('leave_lobby', { roomId });
+      socket.off('update_players');
+      socket.off('start_game');
+    };
+  }, [socket, user, roomId]);
+
+  const startGame = () => {
+    if (players.length >= 2) {
+      socket.emit('start_game', roomId);
+    }
+  };
+  return (
+    <div className='min-h-screen bg-gray-800 text-white'>
+      {/* Header: Player Info */}
+      <header className='flex items-center justify-between bg-gray-900 p-4'>
+        {session ? (
+          <div className='flex items-center'>
+            <Image
+              src={user?.image ?? '/images/default-profile.png'}
```

Comment:
don't forget to add fallback profile image 
<img width="57" alt="image" src="https://github.com/user-attachments/assets/3f9b4c26-567f-41fd-bb3c-502b8e7d5f4f">